PyPDF2/_cmap.py

```diff
@@ -10,41 +10,106 @@
 # code freely inspired from @twiggy ; see #711
 def build_char_map(
     font_name: str, space_width: float, obj: DictionaryObject
-) -> Tuple[str, float, Dict[int, str], Dict]:
+) -> Tuple[
+    str, float, Union[str, Dict[int, str]], Dict
+]:  # font_type,space_width /2, encoding, cmap
     ft: DictionaryObject = obj["/Resources"]["/Font"][font_name]  # type: ignore
     font_type: str = cast(str, ft["/Subtype"])
 
     space_code = 32
     encoding, space_code = parse_encoding(ft, space_code)
     map_dict, space_code = parse_to_unicode(ft, space_code)
+    # encoding can be either a string for decode (on 1,2 or a variable number of bytes) of a char table (for 1 byte only for me)
+    # if empty string, it means it is than encoding field is not present and we have to select the good encoding from cmap input data
+    if encoding == "":
+        if map_dict[-1] == 1:
```

Comment:
```suggestion
        if -1 in map_dict and map_dict[-1] == 1:
```

Reply:
This is triggered by https://corpora.tika.apache.org/base/docs/govdocs1/972/972621.pdf

Reply:
I've coded diffrerently : if -1 not in, we consider charmap also by default